src/traces/funnelarea/plot.js

```diff
@@ -0,0 +1,306 @@
+                    'M' + cx + ',' + cy +
+                    move(pt.TR) +
+                    line(pt.TR, pt.BR) +
+                    line(pt.BR, pt.BL) +
+                    line(pt.BL, pt.TL) +
+                    'Z';
+
+                slicePath.attr('d', shape);
+
+                // add text
+                var textPosition = pieHelpers.castOption(trace.textposition, pt.pts);
+                var sliceTextGroup = sliceTop.selectAll('g.slicetext')
+                    .data(pt.text && (textPosition !== 'none') ? [0] : []);
+
+                sliceTextGroup.enter().append('g')
+                    .classed('slicetext', true);
+                sliceTextGroup.exit().remove();
+
+                sliceTextGroup.each(function() {
+                    var sliceText = Lib.ensureSingle(d3.select(this), 'text', '', function(s) {
+                        // prohibit tex interpretation until we can handle
+                        // tex and regular text together
+                        s.attr('data-notex', 1);
+                    });
+
+                    sliceText.text(pt.text)
+                        .attr({
+                            'class': 'slicetext',
+                            transform: '',
+                            'text-anchor': 'middle'
+                        })
+                        .call(Drawing.font, determineInsideTextFont(trace, pt, gd._fullLayout.font))
+                        .call(svgTextUtils.convertToTspans, gd);
+
+                    // position the text relative to the slice
+                    var textBB = Drawing.bBox(sliceText.node());
+                    var transform;
+
+                    var x0, x1;
+                    var y0 = Math.min(pt.BL[1], pt.BR[1]);
+                    var y1 = Math.max(pt.TL[1], pt.TR[1]);
+
+                    x0 = Math.max(pt.TL[0], pt.BL[0]);
+                    x1 = Math.min(pt.TR[0], pt.BR[0]);
+
+                    transform = getTransformToMoveInsideBar(x0, x1, y0, y1, textBB, {
+                        isHorizontal: true,
+                        constrained: true,
+                        angle: 0,
+                        anchor: 'middle'
+                    });
+
+                    sliceText.attr('transform',
+                        'translate(' + cx + ',' + cy + ')' + transform
+                    );
+                });
+            });
+
+            // add the title
+            var titleTextGroup = d3.select(this).selectAll('g.titletext')
+                .data(trace.title.text ? [0] : []);
+
+            titleTextGroup.enter().append('g')
+                .classed('titletext', true);
+            titleTextGroup.exit().remove();
+
+            titleTextGroup.each(function() {
+                var titleText = Lib.ensureSingle(d3.select(this), 'text', '', function(s) {
+                    // prohibit tex interpretation as above
+                    s.attr('data-notex', 1);
+                });
+
+                var txt = fullLayout.meta ?
```

Comment:
We'll need to generalise this to be inline with https://github.com/plotly/plotly.js/pull/3865 - either in this PR or in #3865 depending on which one gets merged first.

Reply:
Taking care of this one!

Reply:
Done during latest master merge in https://github.com/plotly/plotly.js/pull/3876/commits/1426be7b6ea701f31e6e73d65a3190670818ce5c